packages/fxa-auth-server/lib/routes/subscriptions/utils.ts

```diff
@@ -53,3 +54,9 @@ export function handleAuthScoped(auth: AuthRequest['auth'], scopes: string[]) {
   const { user: uid, email } = auth.credentials;
   return { uid, email } as { uid: string; email: string };
 }
+
+export function generateIdempotencyKey(uid: string, priceId?: string) {
+  let sha = createHash('sha256');
+  sha.update(`${uid}${priceId}`);
+  return sha.digest('base64url');
+}
```

Comment:
Just curious: why we are hashing?

Reply:
I suppose mainly it makes it easier to add on additional fields if we want to in future.